### DISCUSSION TITLE
Server isn't running in WSL

### DISCUSSION THREAD

**Discussion by dtdang** [fief-dev/fief#479] [Q&A]
## Describe the bug
We are using the fief client to connect to our fief authorization server. It is supposed to launch a local host server from using the client but it doesn't appear to be working. 
Here is what it is doing:

```
$ silverback login -p staging
⠙ Please complete authentication in your browser./usr/bin/xdg-open: 882: x-www-browser: not found
/usr/bin/xdg-open: 882: firefox: not found
⠹ Please complete authentication in your browser./usr/bin/xdg-open: 882: iceweasel: not found
⠸ Please complete authentication in your browser./usr/bin/xdg-open: 882: seamonkey: not found
⠼ Please complete authentication in your browser./usr/bin/xdg-open: 882: mozilla: not found
⠴ Please complete authentication in your browser./usr/bin/xdg-open: 882: epiphany: not found
/usr/bin/xdg-open: 882: konqueror: not found
⠦ Please complete authentication in your browser./usr/bin/xdg-open: 882: chromium: not found
⠧ Please complete authentication in your browser./usr/bin/xdg-open: 882: chromium-browser: not found
/usr/bin/xdg-open: 882: google-chrome: not found
⠇ Please complete authentication in your browser./usr/bin/xdg-open: 882: www-browser: not found
/usr/bin/xdg-open: 882: links2: not found
⠏ Please complete authentication in your browser./usr/bin/xdg-open: 882: elinks: not found
⠋ Please complete authentication in your browser./usr/bin/xdg-open: 882: links: not found
/usr/bin/xdg-open: 882: lynx: not found
⠙ Please complete authentication in your browser./usr/bin/xdg-open: 882: w3m: not found
xdg-open: no method available for opening 'https://account.apeworx.io/staging/authorize?response_type=code&client_id=zw-JGVu5GsUPyIuW939UbIax-Z7ygqzPKlu6dMqRfKQ&redirect_uri=http%3A%2F%2Flocalhost%3A51562%2Fcallback&scope=openid+offline_access&code_challenge=cnzGFqDcB07Embqv5lhzWAi24UK5_wzF834Yba15eHE&code_challenge_method=S256'
⠧ Please complete authentication in your browser.
```

Server does not seem to be created and there are no warnings if the server isn't created in the process. The xdg-open link works but the callback route can't find the server.

![image](https://github.com/user-attachments/assets/1b9ee4b9-c420-4d45-97a4-76836da181fc)

## Configuration

* Python version:  Python 3.11.1
* Fief client version: 0.20.0

* Fief Cloud or self-hosted: self-hosted
* If self-hosted, Fief version: 0.29.0

## Additional context

I am using WSL 2 on Ubuntu-22.04

**Comment by dtdang** [comment]
I noticed that this doesn't seem to work either in WSL 
```
$ python -m http.server
Serving HTTP on 0.0.0.0 port 8000 (http://0.0.0.0:8000/) ...
```
![image](https://github.com/user-attachments/assets/40756cd5-eebd-4e8d-8eac-f704ff25bcfe)

**Comment by dtdang** [comment]
Followed the instructions [here](https://stackoverflow.com/a/73690504) to get WSL2 Mirrored Network Mode running and was able to get `python -m http.server` running as intended but still doesn't fix the issue with the fief client.

**Comment by fief-bailiff** [comment]
Hail, @dtdang 👋 Welcome to Fief's kingdom!

Our team will get back to you very soon to help.

In the meantime, take a minute to [star our repository](https://github.com/fief-dev/fief) ⭐️

<p align="center">
<img width="656" alt="star-fief" src="https://github.com/fief-dev/fief/assets/1144727/b2db7f59-102f-47e5-9751-67fa10372c52">
</p>

## Want to support us?

Subscribe to one of our paid plan to help us continue our work and receive exclusive information and benefits! Starts at $5/month 🪙

<p align="center">
    <a href="https://polar.sh/fief-dev">
    <picture>
      <source media="(prefers-color-scheme: dark)" srcset="https://polar.sh/embed/subscribe.svg?org=fief-dev&darkmode=1">
      <img alt="Subscribe" src="https://polar.sh/embed/subscribe.svg?org=fief-dev">
    </picture>
    </a>
</p>

Farewell!

**Comment by frankie567** [comment]
Hmm, I see. By default, the CLI will open a webserver on `localhost:51562`. My guess is that WSL2 will only forward sockets that are listening on all addresses, i.e.`0.0.0.0`. Try this:

```py
fief_auth.authorize(("0.0.0.0", 51562))  
```
